lib/stripe/util.rb

```diff
@@ -208,6 +209,14 @@ def self.normalize_opts(opts)
       end
     end
 
+    # Will either retrieve the contents of a URL or return the file_param
+    # if the argument does not satisfy the regex
+    def self.normalize_file_params(file_param)
+      return file_param unless file_param =~ /^http.*/
```

Comment:
This regex is an almost exact example of the various security problems outlined with OpenURI in this blogpost: [Using open-uri? Check your code - you're playing with fire!](https://sakurity.com/blog/2015/02/28/openuri.html). 

It _seems_ that using `URI.parse` (which [happens internally](https://github.com/ruby/ruby/blob/597ec4354333573d69f57dad931cb8ef477d191a/lib/open-uri.rb#L150) anyway for strings) improves safety for processing `file_param` here, but _don't_ take this suggestion for granted as secure (I whipped it up in just a couple of minutes):
```ruby
uri = URI.parse(file_param) # for example 'http:/../../../../../etc/passwd'
return file_param unless /\Ahttps?\z/ =~ uri.scheme && uri.host # false for above example
OpenURI.open_uri(uri)
```

But it still leaves open issues, like following redirects and monkey patching `Kernel#open` for applications using this gem. At this point, I'm not sure this should be included at all, as you might not want your app to download from any random URL or over plain HTTP. I think it's better your app creates a Tempfile for this gem to use after your app's specific logic has decided it's safe to do so.

Reply:
Yikes, thank you for catching that and linking the article! Considering that risks, I agree with you that it'd be better to have the application be responsible for dealing with links.